Source/WebCore/platform/cocoa/VideoFullscreenModelVideoElement.h

```diff
@@ -90,6 +86,20 @@ class VideoFullscreenModelVideoElement : public VideoFullscreenModel, public Eve
     WEBCORE_EXPORT VideoFullscreenModelVideoElement();
 
 private:
+    class VideoListener final : public EventListener {
+    public:
+        static Ref<VideoListener> create(VideoFullscreenModelVideoElement* parent)
```

Comment:
Would be better and cheaper to pass a ref (assuming parent here is probably never null).

Reply:
The parent can die, as this is a listener and is refcounted.

Reply:
Looking at VideoListener::create in this patch, it seems it is given `this` (maybe I missed some code paths though).
If so, it seems we could pass `*this` so that the creation of the weak pointer is made cheaper as this would remove a nullptr check.

Reply:
yes